### ISSUE TITLE
Chebfun2v roots fails on linear example

### ISSUE THREAD

**Issue by ajt60gaibb** [chebfun/chebfun#1505]
I have been playing a lot with `chebfun2v/roots` recently and there is an example where it misses a well-conditioned root: 

```
p = @(x,y) y + x/2 + 1/10; 
q = @(x,y) y - 2.1*x + 2; 
r = roots( chebfun2v( p, q ) )
r =
     []
```

It actually has a solution inside `[-1,1]x[-1,1]`:

```
[ .5 1 ; -2.1 1] \ [.1 ; -2]
ans =
   0.807692307692308
  -0.303846153846154
```

See also: 
```
plot(roots(chebfun2(p))), hold on, plot(roots(chebfun2(q)))
```
![intersection](https://cloud.githubusercontent.com/assets/1814063/7748454/c6bb51cc-ff92-11e4-90a9-4a5db9419084.png)

**Comment by aaustin141** [comment]
Is this perhaps related to #1232?

**Comment by ajt60gaibb** [comment]
Oh, yes. It is!  Thank you for the reminder.  If I do the same trick as in #1232 and change `eps` to  `eps/10`, then `chebfun2v/roots` works on the example above. But, I still do not understand why an `eps` perturbation to the Bezout resultant matrix removes a well-conditioned solution that is in the interior of the domain.

**Comment by aaustin141** [comment]
OK.  In my case, changing `eps/2` to `eps/10` does not work.  Neither does changing it to `eps/100` or even to `0`.  I can only get an answer in this case when I comment out [the offending lines](https://github.com/chebfun/chebfun/blob/9e28b8fa7a2f90f0299acc7f1b61cf5730fe7ed6/%40chebfun2v/roots.m#L385-389) entirely.  My `B` array just prior to executing these lines is the following:
```
B(:,:,1) =
     0
B(:,:,2) =
   0.318627451044118
B(:,:,3) =
   0.148284313740319
```
Immediately after executing those lines, it is
```
B(:,:,1) =
   0.318627451044118
B(:,:,2) =
   0.148284313740319
```
as you would expect, based on what the code there is doing.  So it seems that the presence or absence of that zero element matters here for some reason.  If I comment out exactly the lines I indicated, I get
```
>> p = @(x, y) y + x/2 + 1/10;
>> q = @(x, y) y - 2.1*x + 2;
>> r = roots(chebfun2v(p, q)) 
r =
   0.730769230769231  -0.465384615384615
```
This differs from the value that you posted above, which I suspect was for a different example, as substitution verifies that this is indeed a root for the system:
```
>> p(r(1), r(2))
ans =
    -2.775557561562891e-17
>> q(r(1), r(2))
ans =
     0
```
Any ideas?

**Comment by aaustin141** [comment]
I think I've finally figured this out.  It has nothing to do with getting rid of negligible components of the `B` array.  Instead, it's a bug in the part of the `chebT1rtsmatgep` subroutine that handles the linear case, namely [this line](https://github.com/chebfun/chebfun/blob/9e28b8fa7a2f90f0299acc7f1b61cf5730fe7ed6/%40chebfun2v/roots.m#L639).  I think the arguments to `eig` here have been reversed; the line should really be `r = eig(c(:,:,2),-c(:,:,1));`.

The reason that commenting out the lines that eliminate small leading entries in `B` appears to fix things is that for many bivariate linear problems, prior to executing those lines, `size(B, 3)` will be 3, and `B(:, :, 1)` will be very small.  Executing those lines causes this small component to get stripped, so that when `B` is  passed into `chebT1rtsmatgep` later, it has `size(B, 3)` equal to 2, which triggers the handler for the linear case.  If, on the other hand, those lines are skipped, the small coefficient doesn't get stripped, so the argument to `chebT1rtsmatgep` has size 3 in the third dimension, and the general handler, which does things correctly, gets called instead.

Note that this explanation is consistent with the results I observed in my previous post, where `B(:, :, 1)` is exactly zero on my machine for this example.  That this happened was very fortunate, as it strongly hinted that the real problem was elsewhere.  I was eventually able to discover what was wrong by trying a whole bunch of randomly selected bivariate linear examples and finding that `chebfun2v` missing the solutions was the rule rather than the exception.  With the changes in I just made to address this in #1508, I now find that it nails these problems every time.

Sorry for leading everyone on a wild goose chase regarding this.  If I'd been more attentive when looking at #1232, we probably would have found this sooner.

Note that this bug also exists in Chebfun v4.

@ajt60gaibb, would you be willing to take a look at the fixes I've made in #1508 and let me know if I'm missing anything else here?  I've gone ahead and returned the cutoff tolerance for small components of `B` to its former value of `10*eps` since that was, in fact, never the problem to begin with.  I've also added a test for the exact example used here so we don't mess this up again in the future.
